diff --git a/thrift/lib/cpp2/server/ThriftServer.h b/thrift/lib/cpp2/server/ThriftServer.h
index 987a2ee5a..c320f3882 100644
--- a/thrift/lib/cpp2/server/ThriftServer.h
+++ b/thrift/lib/cpp2/server/ThriftServer.h
@@ -862,6 +862,6 @@ class ThriftServer : public apache::thrift::BaseThriftServer,
 template <typename AcceptorClass, typename SharedSSLContextManagerClass>
 class ThriftAcceptorFactory : public wangle::AcceptorFactorySharedSSLContext {
  public:
-  ThriftAcceptorFactory<AcceptorClass, SharedSSLContextManagerClass>(
+  ThriftAcceptorFactory(
       ThriftServer* server)
       : server_(server) {}
